### PR TITLE
Add ticksToDays, ticksToHours, ticksToMinutes pre-built functions

### PR DESCRIPTION
**Pull request by cosmicshuai**
fixes: #3989
Add ticksToDays, ticksToHours, ticksToMinutes
ticksToDays is calculated by
Math.Floor(ticks/864000000000)

ticksToHours is calculated by
Math.Floor(ticks/36000000000)

ticksToMinutes is calculated by
Math.Floor(ticks/600000000)

### CODE REVIEW

**Comment by cosmicshuai** [comment]
@vishwacsena 
Do you have any idea or suggestions for these three functions?
One of my concern is the returned value should be an integer(after a math.floor operation) or just return a double value( without a math.floor operation)?

**Comment by vishwacsena** [comment]
let's return a double value. The user should then be able to do a floor once we support that as well as via prebuilt functions.

File: libraries/AdaptiveExpressions/ExpressionFunctions.cs [hunk 1]
```diff
@@ -3762,6 +3762,81 @@ private static IEnumerable<object> Flatten(IEnumerable<object> list, int dept)
...
+                            else
+                            {
+                                error = $"{expr} should contain an integer of ticks";
+                            }
+                        }
+
+                        return (value, error);
+                    },
+                    ReturnType.Number,
+                    expr => ValidateArityAndAnyType(expr, 1, 1, ReturnType.Number)),
```
Comment (Danieladu) [inline libraries/AdaptiveExpressions/ExpressionFunctions.cs:3789]: We could add a new method called "ValidateUnaryNumber"

Comment (cosmicshuai) [reply libraries/AdaptiveExpressions/ExpressionFunctions.cs]: ValidateUnaryNumber added

File: libraries/AdaptiveExpressions/ExpressionFunctions.cs [hunk 2]
```diff
@@ -3762,6 +3769,81 @@ private static IEnumerable<object> Flatten(IEnumerable<object> list, int dept)
...
                     ReturnType.Number,
                     expr => ValidateArityAndAnyType(expr, 1, 1, ReturnType.String)),
+                new ExpressionEvaluator(
+                    ExpressionType.TicksToDays,
+                    (expr, state, options) =>
+                    {
+                        object value = null;
+                        string error = null;
+                        IReadOnlyList<object> args;
+                        long ticksPerDay = 864000000000;
```
Comment (chrimc62) [inline libraries/AdaptiveExpressions/ExpressionFunctions.cs:3779]: >864000000000 [](start = 43, length = 12)

Don't you need an L on the end?  This is bigger than an int.  Should also be a constant and not local.

File: libraries/AdaptiveExpressions/ExpressionFunctions.cs [hunk 3]
```diff
@@ -3762,6 +3769,81 @@ private static IEnumerable<object> Flatten(IEnumerable<object> list, int dept)
...
+                            else
+                            {
+                                error = $"{expr} should contain an integer of ticks";
+                            }
+                        }
+
+                        return (value, error);
+                    },
+                    ReturnType.Number,
+                    expr => ValidateUnaryNumber(expr)),
```
Comment (Danieladu) [inline libraries/AdaptiveExpressions/ExpressionFunctions.cs:3796]: just `ValidateUnaryNumber` is ok, no need to apply the whole method with `expr => ValidateUnaryNumber(expr)`

File: libraries/AdaptiveExpressions/ExpressionFunctions.cs (hunk 2 above)
Comment (cosmicshuai) [reply libraries/AdaptiveExpressions/ExpressionFunctions.cs]: changed

File: libraries/AdaptiveExpressions/ExpressionFunctions.cs (hunk 3 above)
Comment (cosmicshuai) [reply libraries/AdaptiveExpressions/ExpressionFunctions.cs]: changed

**Pull request merged**